backend/rdf4j/src/main/java/org/dotwebstack/framework/backend/rdf4j/query/StaticQueryFetcher.java

```diff
@@ -55,10 +77,48 @@ public Object get(@NonNull DataFetchingEnvironment environment) {
         .orElseGet(TreeModel::new);
   }
 
+  private Object getQueryResultAsModel(@NonNull DataFetchingEnvironment environment,
+      GraphQLFieldDefinition fieldDefinition) {
+    GraphQueryResult graphQueryResult = fetchGraphQuery(environment, fieldDefinition, staticSparqlQuery);
+
+    Model model = mapQueryResultToModel(graphQueryResult);
+
+    if (model.isEmpty()) {
+      return null;
+    }
+
+    return model;
+  }
+
+  private Object getQueryResultAsIri(@NonNull DataFetchingEnvironment environment,
+      GraphQLFieldDefinition fieldDefinition) {
+    TupleQueryResult tupleQueryResult = fetchTupleQuery(environment, fieldDefinition, staticSparqlQuery);
+
+    BindingSet queryBindingSet = tupleQueryResult.next();
+
+    validateQueryHasOneResult(tupleQueryResult, staticSparqlQuery);
+
+    String bindingName = queryBindingSet.getBindingNames()
+        .stream()
+        .findFirst()
+        .get();
+
+    return queryBindingSet.getValue(bindingName);
+  }
+
+  private void validateQueryHasOneResult(TupleQueryResult tupleQueryResult, String query) {
+    if (tupleQueryResult.hasNext()) {
+      throw illegalStateException("Query result for query: {} has more than 1 result", query);
+    }
+  }
+
   public static boolean supports(GraphQLFieldDefinition fieldDefinition) {
-    return Rdf4jScalars.MODEL.getName()
-        .equals(GraphQLTypeUtil.unwrapAll(fieldDefinition.getType())
-            .getName());
+    String inputTypeName = GraphQLTypeUtil.unwrapAll(fieldDefinition.getType())
+        .getName();
+
+    return Stream.of(Rdf4jScalars.IRI, Rdf4jScalars.MODEL)
```

Comment:
static set SUPPORTED_TYPE_NAMES maken en dan

```suggestion
    return SUPPORTED_TYPE_NAMES.contains(inputTypeName)
```
?